lib/ferrum/page/screenshot.rb

```diff
@@ -42,12 +42,7 @@ def pdf(**opts)
         path, encoding = common_options(**opts)
         options = pdf_options(**opts).merge(transferMode: "ReturnAsStream")
         handle = command("Page.printToPDF", **options).fetch("stream")
-
-        if path
-          stream_to_file(handle, path: path)
-        else
-          stream_to_memory(handle, encoding: encoding)
-        end
+        Utils::Stream.for(self).fetch(handle, encoding: encoding, path: path)
```

Comment:
These classes should be loosely coupled, and it is definitely not a good idea to pass instance of page to utils. We better instantiate utils stream and yield IO with the block so that we don't pass page and still loop through the stream:

```ruby
Utils::Stream.fetch(handle, encoding: encoding, path: path) do
  command("IO.read", handle: handle, size: STREAM_CHUNK)
end
```